File: backend/scam-g18/src/main/java/es/codeurjc/scam_g18/dto/ProfileDTO.java
package es.codeurjc.scam_g18.dto;

import java.util.List;
import java.util.Map;
import java.util.Set;

public record ProfileDTO(
        Long id,
        String username,
        String email,
        String country,
        String shortDescription,
        String currentGoal,
        String weeklyRoutine,
        String comunity,
        String profileImage,
        boolean isProfileOwner,
        String userType,
        int completedCourses,
        List<String> completedCourseNames,
        int inProgressCount,
        Set<String> userTags,
        List<Map<String, Object>> subscribedCourses,
        List<Map<String, Object>> userEvents,
        int averageProgress,
        int totalEnrollments,
        long totalLessonsCompleted,
        long completedLessonsThisMonth,
        String averageLessonsPerMonth,
        List<Map<String, Object>> createdCourses,
        boolean hasMultipleCourses) {
}

File: frontend/scam-g18/app/dtos/ProfileDTO.ts
export type ProfileDTO = {
  id: number;
  username: string;
  email: string;
  country: string | null;
  shortDescription: string | null;
  currentGoal: string | null;
  weeklyRoutine: string | null;
  comunity: string | null;
  profileImage: string | null;
  profileOwner: boolean;
  userType: string;
  completedCourses: number;
  completedCourseNames: string[];
  inProgressCount: number;
  userTags: string[];
  subscribedCourses: Array<Record<string, unknown>>;
  userEvents: Array<Record<string, unknown>>;
  averageProgress: number;
  totalEnrollments: number;
  totalLessonsCompleted: number;
  completedLessonsThisMonth: number;
  averageLessonsPerMonth: string;
  createdCourses: Array<Record<string, unknown>>;
  hasMultipleCourses: boolean;
};

export type ProfileUpdateDTO = {
  username: string;
  email: string;
  country?: string | null;
  shortDescription?: string | null;
  currentGoal?: string | null;
  weeklyRoutine?: string | null;
  comunity?: string | null;
};
